File: app/src/main/java/com/dpis/module/applist/InstalledAppCatalogItem.java
package com.dpis.module.applist;

import android.content.pm.ApplicationInfo;
import android.graphics.drawable.Drawable;

public final class InstalledAppCatalogItem {
    public final String label;
    public final String packageName;
    public final boolean systemApp;
    public final boolean hyperOsNativeProxyCandidate;
    final ApplicationInfo applicationInfo;
    public volatile Drawable icon;

    public InstalledAppCatalogItem(String label,
            String packageName,
            boolean systemApp,
            boolean hyperOsNativeProxyCandidate,
            ApplicationInfo applicationInfo,
            Drawable icon) {
        this.label = label;
        this.packageName = packageName;
        this.systemApp = systemApp;
        this.hyperOsNativeProxyCandidate = hyperOsNativeProxyCandidate;
        this.applicationInfo = applicationInfo;
        this.icon = icon;
    }
}

File: app/src/main/java/com/dpis/module/applist/InstalledAppCatalogItem.kt
package com.dpis.module.applist

import android.content.pm.ApplicationInfo
import android.graphics.drawable.Drawable

class InstalledAppCatalogItem(
    @JvmField val label: String,
    @JvmField val packageName: String,
    @JvmField val systemApp: Boolean,
    @JvmField val hyperOsNativeProxyCandidate: Boolean,
    internal val applicationInfo: ApplicationInfo,
    @JvmField @Volatile var icon: Drawable?,
    @JvmField val firstInstallTime: Long,
    @JvmField val lastUpdateTime: Long,
    @JvmField val labelResolved: Boolean = false,
) {
    fun withResolvedLabel(label: String): InstalledAppCatalogItem =
        InstalledAppCatalogItem(
            label,
            packageName,
            systemApp,
            hyperOsNativeProxyCandidate,
            applicationInfo,
            icon,
            firstInstallTime,
            lastUpdateTime,
            labelResolved = true,
        )
}
